lib/cog/commands/echo.ex

```diff
@@ -8,5 +8,15 @@ defmodule Cog.Commands.Echo do
   rule "when command is #{Cog.Util.Misc.embedded_bundle}:echo allow"
 
   def handle_message(req, state),
-    do: {:reply, req.reply_to, Enum.join(req.args, " "), state}
+    do: {:reply, req.reply_to, Enum.map_join(req.args, " ", &serialize/1), state}
+
+  # We should only have to worry about maps and lists here. Anything else
+  # we might get should implement 'to_string'.
+  defp serialize(map) when is_map(map),
+    do: Poison.encode!(map)
+  defp serialize(list) when is_list(list),
```

Comment:
If you change the signature to 

```elixir
serialize(arg) when is_map(arg) or is_list(arg)
```
you can do it in one function head instead of two. It also makes the explanatory comments above unnecessary, since they are saying exactly what the code is saying.